### PR TITLE
Pull request for Sean's updates

### PR DESCRIPTION
**Pull request by srmelody**
Team,
I finished dusting off Leeroy's code and opened a PR.  Please check out the ```sean-notes.md``` for more detail about what changed.  I'll annotate the PR so the team can see my changes.  🎆 
Sean

### CODE REVIEW

File: js/TexttestFixture.html [hunk 1]
```diff
@@ -6,9 +6,10 @@
 
 <script type="text/javascript" src="lib/jquery-1.7.1/jquery-1.7.1.js"></script>
 <script type="text/javascript" src="src/gilded_rose.js"></script>
-
+<link rel="stylesheet" type="text/css" href="gilded-rose.css">
 <script type="text/javascript">
-  $( document ).ready(function() {
+
+  $( document ).ready( () => {
```
Comment (srmelody) [inline js/TexttestFixture.html]: new ES6 arrow functions - used throughout

File: js/TexttestFixture.html [hunk 2]
```diff
@@ -20,31 +21,35 @@
     items.push(new Item('Backstage passes to a TAFKAL80ETC concert', 15, 20));
     items.push(new Item('Backstage passes to a TAFKAL80ETC concert', 10, 49));
     items.push(new Item('Backstage passes to a TAFKAL80ETC concert', 5, 49));
-    // this conjured item does not work properly yet
     items.push(new Item('Conjured Mana Cake', 3, 6));
+    items.push(new Item('Luck Potion, Conjured by Hagrid', 3, 49));
 
-    var days = 2;
+    const days = 20;
```
Comment (srmelody) [inline js/TexttestFixture.html]: const ftw!

File: js/TexttestFixture.html [hunk 3]
```diff
@@ -20,31 +21,35 @@
...
 
-    for (var i = 0; i < days; i++) {
+    for (let i = 0; i < days; i++) {
```
Comment (srmelody) [inline js/TexttestFixture.html]: avoids hoisting

File: js/TexttestFixture.html [hunk 4]
```diff
@@ -20,31 +21,35 @@
...
-        var item = items[j];
-        $('body').append('<p>' + item.name + ', ' + item.sell_in + ', ' + item.quality + '</p>');
-      }
-      $('body').append('<br />');
-    }
-
   });
+
+  const showHeaderFor = (day) => {
+    $('body').append(`<p>-------- day ${day} --------</p><table>`);
```
Comment (srmelody) [inline js/TexttestFixture.html]: We're trying to use interpolated strings now - talk to me if this isn't clear.

File: js/TexttestFixture.html [hunk 5]
```diff
@@ -20,31 +21,35 @@
...
+  }
+
+  const showItemsFor = (day) => {
+    $('body').append('<tr><th>name</th><th>sellIn</th><th>quality</th></tr>');
```
Comment (srmelody) [inline js/TexttestFixture.html]: I changed this to tables but we should talk about whether we want [Flexbox](https://css-tricks.com/snippets/css/a-guide-to-flexbox/) going forward.

File: js/sean-notes.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
+# Readme JS
```
Comment (srmelody) [inline js/sean-notes.md]: important README!

File: js/sean-notes.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+* Collapsing ifs / end its - goal here was to make the main logic functions mostly words for readability and reduce cyclomatic complexity
+* Replaced item.quality - item.quality with 0
+* Restructure the past sell in logic so it was clearer that aged brie was the edge case - the switch statement helps readability and made it easier to add conjured items.
+* Added strict equality (we should lint for this)
+* Added test drivers to allow multiple item scenarios
+* Removed some duplicate logic with guarded increment and decrement functions
+* Added support for Conjured items
+
+# TODO
+* Can we talk to the business stake holders and update the README.md if brie is *not* supposed to increase in quality by 2 past the sell in date?
```
Comment (srmelody) [inline js/sean-notes.md]: At our next team grooming, let's talk through these and get cards on the board for what's next.

File: js/src/gilded_rose.js [hunk 8]
```diff
@@ -6,50 +12,97 @@ function Item(name, sell_in, quality) {
...
+    decrementSellIn(item);
+
+    if (item.sell_in < 0) {
+      handlePastSellIn(item);
     }
+  });
+}
+
+// Normal items are those that increment before the sell_in date and decrement after the sell_in date.
+const handleNormalItem = (item) => {
```
Comment (srmelody) [inline js/src/gilded_rose.js]: I used the term Normal - not trying to imply a value judgment, so I'm open to name suggestions.
